ledger/simulation/trace.go

```diff
@@ -163,29 +176,49 @@ func makeSimulationResult(lastRound basics.Round, request Request, developerAPI
 	}, nil
 }
 
+// TealValue Represents a TEAL value.
+type TealValue struct {
+	_struct struct{} `codec:",omitempty,omitemptyarray"`
+
+	Type  basics.TealType `codec:"type"`
+	Bytes string          `codec:"bytes"`
+	Uint  uint64          `codec:"uint"`
+}
+
```

Comment:
Can we use the existing `basics.TealValue` type instead of introducing a new, slightly different one?
It will already have some useful conversion routines. The only difference will be different names for the fields. I think we can live with the existing ones.

Reply:
Since we already realized we cannot directly adapt to `basics.TealValue` for the `codec` field has succinct name:

>  model.TealValue type that uses the "long" field names (type, uint, bytes), and we probably want to be consistent.

we can adapt to `model.TealValue` in `OpcodeTraceUnit`, by `ToTealValue` to cast `stackValue` to `basics.TealValue`, then directly assign each field to `model.TealValue`, as you suggested in earlier comment, e.g.,

```golang
		tealValue := cx.Stack[i].ToTealValue()
		o.Added = append(o.Added, model.TealValue{
			Type:  uint64(tealValue.Type),
			Uint:  tealValue.Uint,
			Bytes: tealValue.Bytes,
		})
```